tempesta_fw/t/unit/test_sched_http.c

```diff
@@ -0,0 +1,300 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2015 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include "http_msg.h"
+
+#include "helpers.h"
+#include "sched_helper.h"
+#include "test.h"
+#include "cfg.h"
+#include "kallsyms_helper.h"
+
+/* Export syms*/
+static TfwScheduler *(*tfw_sched_lookup_ptr)(const char *name);
+static void (*spec_cleanup_ptr)(TfwCfgSpec specs[]);
+
+static int
+parse_cfg(const char *cfg_text)
+{
+	struct list_head mod_list;
+	TfwCfgMod cfg_mod;
+
+	cfg_mod = *tfw_cfg_mod_lookup("tfw_sched_http");
+	INIT_LIST_HEAD(&cfg_mod.list);
+	INIT_LIST_HEAD(&mod_list);
+	list_add(&cfg_mod.list, &mod_list);
+
+	return tfw_cfg_parse_mods_cfg(cfg_text, &mod_list);
+}
+
+static void
+cleanup_cfg(void)
+{
+	TfwCfgMod cfg_mod;
+
+	cfg_mod = *tfw_cfg_mod_lookup("tfw_sched_http");
+	spec_cleanup_ptr(cfg_mod.specs);
+}
+
+static void
+test_req(char *req_str, TfwSrvConnection *expect_conn)
```

Comment:
`req_str` is of type `char *` here, but some calls of `test_req` later use string literals, which are in fact, `const`.
Since `tfw_http_parse_req` can write to `req_str`, that can fail.